test/e2e/framework.go

```diff
@@ -1142,6 +1148,38 @@ func (data *TestData) GetEncapMode() (config.TrafficEncapModeType, error) {
 	return config.TrafficEncapModeInvalid, fmt.Errorf("antrea-conf config map is not found")
 }
 
+func (data *TestData) getFeatures(confName string, antreaNamespace string) (featuregate.FeatureGate, error) {
+	featureGate := features.DefaultMutableFeatureGate.DeepCopy()
+	cfgMap, err := data.GetAntreaConfigMap(antreaNamespace)
+	if err != nil {
+		return nil, err
+	}
+	var cfg interface{}
+	if err := yaml.Unmarshal([]byte(cfgMap.Data[confName]), &cfg); err != nil {
+		return nil, err
+	}
+	rawFeatureGateMap, ok := cfg.(map[interface{}]interface{})["featureGates"]
+	if !ok || rawFeatureGateMap == nil {
+		return featureGate, err
```

Comment:
nit: err is from `yaml.Unmarshal` and might be confusing here, it should return `nil` explicitly?
and do we need to check both of ok and rawFeatureGateMap to go this branch?

Reply:
Updated.
If no feature is enabled, the map would be nil and then we don't need further application. And the `ok` only indicates whether the key exists or not.